File: unknownbank_flutter/lib/screens/credit_card_detail_screen.dart
import 'package:flutter/material.dart';

import '../models/credit_card_product.dart';
import '../theme/app_colors.dart';

class CreditCardDetailScreen extends StatelessWidget {
  final CreditCardProduct product;
  final VoidCallback onApplyNow;

  const CreditCardDetailScreen({
    super.key,
    required this.product,
    required this.onApplyNow,
  });

  @override
  Widget build(BuildContext context) {
    final heroImage = product.heroImage ?? product.image;

    return Scaffold(
      backgroundColor: Colors.white,
      body: Stack(
        children: [
          Positioned(
            top: 0,
            left: 0,
            right: 0,
            child: SizedBox(
              height: 420,
              child: Image.asset(
                'assets/images/$heroImage',
                fit: product.heroImage != null
                    ? BoxFit.cover
                    : BoxFit.contain,
              ),
            ),
          ),

          SafeArea(
            child: SingleChildScrollView(
              padding: const EdgeInsets.only(
                bottom: 100,
              ),
              child: Column(
                children: [
                  SizedBox(
                    height: 56,
                    child: Align(
                      alignment: Alignment.centerLeft,
                      child: Padding(
                        padding: const EdgeInsets.only(
                          left: 8,
                        ),
                        child: IconButton(
                          onPressed: () {
                            Navigator.of(context).pop();
                          },
                          icon: const Icon(
                            Icons.arrow_back,
                            color: AppColors.textPrimary,
                          ),
                        ),
                      ),
                    ),
                  ),

                  const SizedBox(height: 5),

                  Padding(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 20,
                    ),
                    child: Text(
                      product.name,
                      textAlign: TextAlign.center,
                      style: const TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                        color: Color(0xFF333333),
                      ),
                    ),
                  ),

                  const SizedBox(height: 12),

                  Padding(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 30,
                    ),
                    child: Row(
                      children: [
                        Expanded(
                          child: _InfoColumn(
                            title: 'Min. Income',
                            value: product.minIncome,
                          ),
                        ),

                        Container(
                          width: 1,
                          height: 31,
                          color: const Color(0xFFCFCFCF),
                        ),

                        Expanded(
                          child: _InfoColumn(
                            title: 'Annual Fee',
                            value: product.annualFee,
                          ),
                        ),
                      ],
                    ),
                  ),

                  const SizedBox(height: 245),

                  Padding(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 20,
                    ),
                    child: Text(
                      product.description,
                      textAlign: TextAlign.center,
                      style: const TextStyle(
                        fontSize: 15,
                        height: 1.47,
                        color: Color(0xFF444444),
                      ),
                    ),
                  ),

                  const SizedBox(height: 34),

                  const Text(
                    'Featured Benefits',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.bold,
                      color: Color(0xFF333333),
                    ),
                  ),

                  const SizedBox(height: 18),

                  Padding(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 16,
                    ),
                    child: Row(
                      children: [
                        for (int i = 0;
                            i <
                                product
                                    .featuredBenefits.length;
                            i++) ...[
                          Expanded(
                            child: _BenefitCard(
                              benefit: product
                                  .featuredBenefits[i],
                            ),
                          ),

                          if (i !=
                              product.featuredBenefits
                                      .length -
                                  1)
                            const SizedBox(width: 10),
                        ],
                      ],
                    ),
                  ),
                ],
              ),
            ),
          ),

          Positioned(
            left: 16,
            right: 16,
            bottom: 16,
            child: SafeArea(
              top: false,
              child: SizedBox(
                height: 46,
                child: FilledButton(
                  onPressed: onApplyNow,
                  style: FilledButton.styleFrom(
                    backgroundColor:
                        AppColors.primary,
                    shape: RoundedRectangleBorder(
                      borderRadius:
                          BorderRadius.circular(100),
                    ),
                  ),
                  child: const Text(
                    'Apply Now',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _InfoColumn extends StatelessWidget {
  final String title;
  final String value;

  const _InfoColumn({
    required this.title,
    required this.value,
  });

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Text(
          title,
          style: const TextStyle(
            fontSize: 12,
            color: Color(0xFF777777),
          ),
        ),
        const SizedBox(height: 4),
        Text(
          value,
          textAlign: TextAlign.center,
          style: const TextStyle(
            fontSize: 12,
            fontWeight: FontWeight.bold,
            color: Color(0xFF333333),
          ),
        ),
      ],
    );
  }
}

class _BenefitCard extends StatelessWidget {
  final CreditCardBenefit benefit;

  const _BenefitCard({
    required this.benefit,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 148,
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(10),
        border: Border.all(
          color: const Color(0xFFEEEEEE),
        ),
      ),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Image.asset(
            'assets/images/${benefit.image}',
            width: 52,
            height: 52,
          ),
          const SizedBox(height: 18),
          Text(
            benefit.text,
            textAlign: TextAlign.center,
            style: const TextStyle(
              fontSize: 13,
              height: 1.23,
              color: Color(0xFF333333),
            ),
          ),
        ],
      ),
    );
  }
}

File: Unknownbank/app/src/main/java/com/kasatria/kasatriaunknownbank/CreditCardDetailScreen.kt
package com.kasatria.kasatriaunknownbank

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

import com.kasatria.kasatriaunknownbank.ui.theme.PrimaryBlue
import com.kasatria.kasatriaunknownbank.ui.theme.TextPrimary
import com.kasatria.kasatriaunknownbank.ui.theme.White


@Composable
fun CreditCardDetailScreen(
    onBack: () -> Unit,
    onApplyNow: () -> Unit,
    product: CreditCardProduct
) {

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.White)
    ) {

        /*
         * =================================
         * HERO ARTWORK
         * Figma node 1:383
         * =================================
         */

        val heroImage =
            product.heroImageRes
                ?: product.imageRes

        Image(
            painter = painterResource(
                id = heroImage
            ),
            contentDescription = product.name,
            modifier = Modifier
                .fillMaxWidth()
                .height(420.dp),
            contentScale =
                if (product.heroImageRes != null) {
                    ContentScale.Crop
                } else {
                    ContentScale.Fit
                }
        )


        /*
         * =================================
         * TOP CONTENT
         * =================================
         */

        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
        ) {

            /*
             * Header / Back
             */

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
            ) {

                IconButton(
                    onClick = onBack,
                    modifier = Modifier
                        .padding(start = 8.dp)
                        .align(Alignment.CenterStart)
                        .size(48.dp)
                ) {

                    Icon(
                        imageVector =
                            Icons.AutoMirrored.Filled.ArrowBack,

                        contentDescription = "Back",

                        tint = TextPrimary
                    )
                }
            }


            /*
             * Card name
             */

            Text(
                text = product.name,

                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 5.dp),

                color = Color(0xFF333333),

                fontSize = 18.sp,

                fontWeight = FontWeight.Bold,

                textAlign = TextAlign.Center
            )


            /*
             * Income + Annual Fee
             */

            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(
                        start = 30.dp,
                        end = 30.dp,
                        top = 12.dp
                    ),

                verticalAlignment =
                    Alignment.CenterVertically
            ) {

                Column(
                    modifier =
                        Modifier.weight(1f),

                    horizontalAlignment =
                        Alignment.CenterHorizontally
                ) {

                    Text(
                        text = "Min. Income",
                        color = Color(0xFF777777),
                        fontSize = 12.sp
                    )

                    Spacer(
                        modifier =
                            Modifier.height(4.dp)
                    )

                    Text(
                        text = "RM 100,000 per annum",
                        color = Color(0xFF333333),
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Bold
                    )
                }


                Box(
                    modifier = Modifier
                        .width(1.dp)
                        .height(31.dp)
                        .background(
                            Color(0xFFCFCFCF)
                        )
                )


                Column(
                    modifier =
                        Modifier.weight(1f),

                    horizontalAlignment =
                        Alignment.CenterHorizontally
                ) {

                    Text(
                        text = "Annual Fee",
                        color = Color(0xFF777777),
                        fontSize = 12.sp
                    )

                    Spacer(
                        modifier =
                            Modifier.height(4.dp)
                    )

                    Text(
                        text = "FREE*",
                        color = Color(0xFF333333),
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Bold
                    )
                }
            }


            /*
             * Hero image occupies until ~420dp.
             */

            Spacer(
                modifier =
                    Modifier.height(245.dp)
            )


            /*
             * =================================
             * DESCRIPTION
             * =================================
             */

            Text(
                text =
                    "Enjoy instant reward redemptions with\n" +
                            "Premier Travel Credit Card! Get up to 5x reward\n" +
                            "points, complimentary travel insurance and annual\n" +
                            "passes to selected airport lounges.",

                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp),

                color = Color(0xFF444444),

                fontSize = 15.sp,

                lineHeight = 22.sp,

                textAlign = TextAlign.Center
            )


            Spacer(
                modifier =
                    Modifier.height(34.dp)
            )


            /*
             * =================================
             * FEATURED BENEFITS
             * =================================
             */

            Text(
                text = "Featured Benefits",

                modifier =
                    Modifier.fillMaxWidth(),

                color = Color(0xFF333333),

                fontSize = 15.sp,

                fontWeight =
                    FontWeight.Bold,

                textAlign =
                    TextAlign.Center
            )


            Spacer(
                modifier =
                    Modifier.height(18.dp)
            )


            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),

                horizontalArrangement =
                    Arrangement.spacedBy(10.dp)
            ) {

                BenefitCard(
                    modifier =
                        Modifier.weight(1f),

                    icon =
                        R.drawable.benefit_fee_waiver,

                    text =
                        "Annual Fee\nWaiver"
                )


                BenefitCard(
                    modifier =
                        Modifier.weight(1f),

                    icon =
                        R.drawable.benefit_cashback,

                    text =
                        "Up to 15%\nCashback"
                )


                BenefitCard(
                    modifier =
                        Modifier.weight(1f),

                    icon =
                        R.drawable.benefit_points,

                    text =
                        "Earn 5x\nPoints"
                )
            }
        }


        /*
         * =================================
         * APPLY NOW
         * =================================
         */

        Button(
            onClick = onApplyNow,

            modifier = Modifier
                .align(Alignment.BottomCenter)
                .navigationBarsPadding()
                .padding(
                    start = 16.dp,
                    end = 16.dp,
                    bottom = 16.dp
                )
                .fillMaxWidth()
                .height(46.dp),

            shape =
                RoundedCornerShape(100.dp),

            colors =
                ButtonDefaults.buttonColors(
                    containerColor = PrimaryBlue,
                    contentColor = White
                )
        ) {

            Text(
                text = "Apply Now",
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold
            )
        }
    }
}


@Composable
private fun BenefitCard(
    icon: Int,
    text: String,
    modifier: Modifier = Modifier
) {

    Column(
        modifier = modifier
            .height(148.dp)
            .background(
                color = White,
                shape = RoundedCornerShape(10.dp)
            )
            .border(
                width = 1.dp,
                color = Color(0xFFEEEEEE),
                shape = RoundedCornerShape(10.dp)
            ),

        horizontalAlignment =
            Alignment.CenterHorizontally,

        verticalArrangement =
            Arrangement.Center
    ) {

        Image(
            painter =
                painterResource(
                    id = icon
                ),

            contentDescription = null,

            modifier =
                Modifier.size(52.dp),

            contentScale =
                ContentScale.Fit
        )


        Spacer(
            modifier =
                Modifier.height(18.dp)
        )


        Text(
            text = text,

            color =
                Color(0xFF333333),

            fontSize = 13.sp,

            lineHeight = 16.sp,

            textAlign =
                TextAlign.Center
        )
    }
}